build-tools/packages/build-tools/src/type-test-generator/compatibility.ts

```diff
@@ -163,3 +162,37 @@ namespace Test7 {
 	// but only in one direction:
 	type _check5 = requireAssignableTo<TypeOnly<Test7.Baz>, TypeOnly<Test6.Foo>>;
 }
+
+namespace Test_TypeOnly_Preserves_Primitives {
+	// The 'null' and 'undefined' values cannot be nominal typed (they are not
+	// valid enum values and intersection ('&') with 'null'/'undefined' -> 'never').
+	// Just verify that the 'null' and 'undefined' values are preserved.
+	type _check_undefined1 = requireAssignableTo<TypeOnly<undefined>, undefined>;
+	type _check_undefined2 = requireAssignableTo<undefined, TypeOnly<undefined>>;
+
+	type _check_null1 = requireAssignableTo<TypeOnly<null>, null>;
+	type _check_null2 = requireAssignableTo<null, TypeOnly<null>>;
+
+	// Types that extend 'number' and 'string' are stripped to their primitive types.
+	// (Nominal typing is erased.)
+	type brandedNumber = number & { brand: "Number" };
+	type _check_number1 = requireAssignableTo<TypeOnly<brandedNumber>, number>;
+	type _check_number2 = requireAssignableTo<number, TypeOnly<brandedNumber>>;
+
+	type brandedString = string & { brand: "String" };
+	type _check_string1 = requireAssignableTo<TypeOnly<brandedString>, string>;
+	type _check_string2 = requireAssignableTo<string, TypeOnly<brandedString>>;
+
+	// Other primitive types are preserved as-is.
+	type brandedBoolean = boolean & { brand: "Boolean" };
```

Comment:
add a test in here for a union, like `string | number | boolean |  BigInt | symbol | undefined | null` and confirm it is preserved.

Reply:
I added a few test variations involving unions.